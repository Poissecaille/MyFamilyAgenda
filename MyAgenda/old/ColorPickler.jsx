import React, {useEffect, useState} from 'react';
import {Button, View} from 'react-native';
import ColorPicker from 'react-native-wheel-color-picker';

export const ColorPickerObject = props => {
  const [currentColor, setCurrentColor] = useState('#fffff');
  return (
    <>
      {props.visible && (
        <View>
          <ColorPicker
            ref={r => {}}
            color={currentColor}
            onColorChangeComplete={color => {
              setCurrentColor(color);
              props.colorChangeHandler(color);
            }}
            thumbSize={40}
            row={false}
            swatchesLast={false}
            sliderHidden={true}
          />
          {/* <Button title="btn-picker" onPress={() => this.picker.revert()} /> */}
        </View>
      )}
    </>
  );
};
