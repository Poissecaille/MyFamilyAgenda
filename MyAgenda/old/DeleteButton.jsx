import React from 'react';
import styled from 'styled-components/native';
import { colors } from '../Colors';
import { RegularText } from '../RegularText';

const ButtonStyle = styled.TouchableOpacity`
  position:absolute;
  align-items: center;
  background-color: ${colors.LightRed};
  width: 20%;
  height:30%;
  left:180px;
  top:30px;
  border-radius: 10px;
  z-index:1
`;

export const DeleteButton = props => {
    return (
        <ButtonStyle onPress={props.onPress}>
            <RegularText value={props.title}></RegularText>
        </ButtonStyle>
    );
};
