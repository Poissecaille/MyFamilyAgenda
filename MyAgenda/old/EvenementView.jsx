import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

export const EvenementView = (props) => {

  return (
    <>
      <View style={styles.centeredView}>
        <Text style={styles.title}> {props.name} (R.S) </Text>
        <Text style={styles.regular}> {props.date} à {props.place} </Text>
        <Text style={styles.regular}> {props.description} </Text>
        <Text style={styles.regular}> Invités : {props.guests}</Text>
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  centeredView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  title: {
    fontSize: 40,
    fontWeight: 'bold',
  },
  regular: {
    fontSize: 25,
    fontWeight: 'lighter',
  }
});