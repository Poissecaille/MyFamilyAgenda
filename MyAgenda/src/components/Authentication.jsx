import React, { useState, createContext } from 'react';
import firebase from 'firebase/compat/app';
import 'firebase/compat/auth';
import 'firebase/compat/firestore';
import { getDatabase, ref, set, push } from 'firebase/database';

export const AuthenticationContext = createContext();
const db = getDatabase();

const loginRequest = (email, password) =>
  firebase.auth().signInWithEmailAndPassword(email, password);


export const AuthenticationContextProvider = ({ children }) => {
  const [isLoading, setIsLoading] = useState(false);
  const [user, setUser] = useState(null);
  const [error, setError] = useState(null);

  function writeUserData(email) {
    const reference = ref(db, 'Users/');
    const newReference = push(reference) // generate new id under Users/
    set(newReference, {
      email: email,
    });
  }

  function createBaseAgendas(email) {
    const reference = ref(db, 'Agendas/');
    let newReference = push(reference); //generate new id under Agendas/
    set(newReference, {
      name: "Mon agenda familial",
      owner: email,
      members: {},
      events: {},
    });
    newReference = push(reference); //generate new id under Agendas/
    set(newReference, {
      name: "Mon agenda personnel",
      owner: email,
      members: {},
      events: {},
    });
  }



  firebase.auth().onAuthStateChanged(usr => {
    if (usr) {
      setUser(usr);
      setIsLoading(false);
    } else {
      setIsLoading(false);
    }
  });

  const onLogin = (email, password) => {
    setIsLoading(true);
    loginRequest(email, password)
      .then(u => {
        setUser(u);
        setIsLoading(false);
      })
      .catch(e => {
        setIsLoading(false);
        setError(e.toString());
      });
  };

  const onRegister = (email, password, repeatedPassword) => {
    setIsLoading(true);
    if (password !== repeatedPassword) {
      setError('Error: Passwords do not match');
      return;
    }
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(u => {
        writeUserData(email);
        createBaseAgendas(email)
        setUser(u);
        setIsLoading(false);
      })
      .catch(e => {
        setIsLoading(false);
        setError(e.toString());
      });
  };

  const onLogout = () => {
    firebase
      .auth()
      .signOut()
      .then(() => {
        setUser(null);
        setError(null);
      });
  };

  return (
    <AuthenticationContext.Provider
      value={{
        isAuthenticated: !!user,
        user,
        isLoading,
        error,
        onLogin,
        onRegister,
        onLogout,
      }}>
      {children}
    </AuthenticationContext.Provider>
  );
};
