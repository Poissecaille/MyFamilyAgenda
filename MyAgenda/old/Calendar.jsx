import React from 'react';
import { useState } from 'react';
import { Calendar } from 'react-native-calendars';
import { LocaleConfig } from 'react-native-calendars';
import { ColorPickerObject } from './ColorPickler';
import { RegularButton } from './RegularButton';
import { colors } from '../src/utility/Global';

export const CalendarObject = () => {
  const [dateString, setDateString] = useState(
    new Date().toISOString().split('T')[0],
  ); // date formatted as 'YYYY-MM-DD' string
  const [picker, setPicker] = useState(false);
  const [color, setColor] = useState(colors.Primary_Color);
  const [markedDate, setMarkedDate] = useState('{}');

  LocaleConfig.locales['fr'] = {
    monthNames: [
      'Janvier',
      'Février',
      'Mars',
      'Avril',
      'Mai',
      'Juin',
      'Juillet',
      'Août',
      'Septembre',
      'Octobre',
      'Novembre',
      'Décembre',
    ],
    monthNamesShort: [
      'Janv.',
      'Févr.',
      'Mars',
      'Avril',
      'Mai',
      'Juin',
      'Juil.',
      'Août',
      'Sept.',
      'Oct.',
      'Nov.',
      'Déc.',
    ],
    dayNames: [
      'Dimanche',
      'Lundi',
      'Mardi',
      'Mercredi',
      'Jeudi',
      'Vendredi',
      'Samedi',
    ],
    dayNamesShort: ['Dim.', 'Lun.', 'Mar.', 'Mer.', 'Jeu.', 'Ven.', 'Sam.'],
    today: "Aujourd'hui",
  };
  LocaleConfig.defaultLocale = 'fr';

  const colorPickerHandler = () => {
    setPicker(!picker);
  };
  const colorChangeHandler = color => {
    setColor(color);
  };
  const updateMarkedDayHandler = date => {
    let tmp;
    if (typeof markedDate == 'string') {
      tmp = JSON.parse(markedDate);
    } else {
      tmp = markedDate;
    }
    if (!(date.dateString in tmp)) {
      tmp[date.dateString] = {
        selected: true,
        selectedColor: color,
        dotColor: color,
      };
    } else {
      tmp[date.dateString] = {
        selected: !tmp[date.dateString].selected,
        selectedColor: color,
        dotColor: color,
      };
    }
    setMarkedDate(JSON.stringify(tmp));
  };

  return (
    <>
      <RegularButton title={'Color picker'} onPress={colorPickerHandler} />
      <Calendar
        style={{
          marginTop: 80,
          padding: 20,
        }}
        // Initially visible month
        initialDate={dateString}
        // Minimum date that can be selected, dates before minDate will be grayed out.
        minDate={dateString}
        // Do not show days of other months in month page. Default = false
        hideExtraDays={true}
        onDayPress={date => {
          //   console.log('selected date', date, color);
          updateMarkedDayHandler(date);
        }}
        // Enable the option to swipe between months. Default = false
        enableSwipeMonths={true}
        markedDates={JSON.parse(markedDate)}
      />
      <ColorPickerObject
        visible={picker}
        colorChangeHandler={colorChangeHandler}
      />
    </>
  );
};
