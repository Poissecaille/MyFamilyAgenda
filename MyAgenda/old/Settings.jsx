import React, { useContext } from 'react';
import { SafeArea } from './SafeArea';
import styled from 'styled-components/native';
import { List, Avatar } from 'react-native-paper';
import { AuthenticationContext } from '../src/components/Authentication';
import { Text } from './Text';
import { Spacer } from './Spacer';
import { theme } from '../src/utility/Global';

const SettingsItem = styled(List.Item)`
  padding: ${theme.space[3]};
`;
const AvatarContainer = styled.View`
  align-items: center;
`;

export const SettingsScreen = () => {
  const { onLogout, user } = useContext(AuthenticationContext);
  return (
    <SafeArea color={theme.themeColors.Primary_Color}>
      <AvatarContainer>
        <Avatar.Icon size={180} icon="human" />
        <Spacer position="top" size="large">
          <Text variant="label">{user.email}</Text>
        </Spacer>
      </AvatarContainer>
      <List.Section>
        <SettingsItem
          style={{ padding: 16 }}
          title="Logout"
          left={props => <List.Icon {...props} color="black" icon="door" />}
          onPress={onLogout}
        />
      </List.Section>
    </SafeArea>
  );
};
