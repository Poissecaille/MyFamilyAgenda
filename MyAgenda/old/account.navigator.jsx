import React from 'react';
import { createStackNavigator } from '@react-navigation/stack';

import { AccountScreen } from '../src/components/Account';
import { LoginScreen } from '../src/components/Login';
import { RegisterScreen } from '../src/components/Register';


const Stack = createStackNavigator();

export const AccountNavigator = () => (
  <Stack.Navigator headerMode="none">
    <Stack.Screen name="Main" component={AccountScreen} />
    <Stack.Screen name="Login" component={LoginScreen} />
    <Stack.Screen name="Register" component={RegisterScreen} />
  </Stack.Navigator>
);
