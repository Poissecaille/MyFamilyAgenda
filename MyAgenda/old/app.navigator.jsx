import React from 'react';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import {Ionicons} from '@expo/vector-icons';
import {Home} from '../../components/Home';
import {SettingsNavigator} from './settings.navigator';

const Tab = createBottomTabNavigator();

const TAB_ICON = {
  Restaurants: 'md-restaurant',
  Map: 'md-map',
  Settings: 'md-settings',
};

const createScreenOptions = ({route}) => {
  const iconName = TAB_ICON[route.name];
  return {
    tabBarIcon: ({size, color}) => (
      <Ionicons name={iconName} size={size} color={color} />
    ),
  };
};

export const AppNavigator = () => (
  <Tab.Navigator
  screenOptions={createScreenOptions}
  tabBarOptions={{
    activeTintColor: 'tomato',
    inactiveTintColor: 'gray',
  }}>
   <Tab.Screen name="Home" component={Home} />

   <Tab.Screen name="Settings" component={SettingsNavigator} />
 </Tab.Navigator>
);
