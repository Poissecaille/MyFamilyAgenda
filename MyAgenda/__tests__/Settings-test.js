import { SettingsScreen } from '../old/Settings';    
import {AuthenticationContext} from '../src/components/Authentication';
import { expect } from '@jest/globals';
import React from 'react';
import {create,act,render} from 'react-test-renderer';
// Note: test renderer must be required after react-native.
jest.useFakeTimers()

jest.mock("firebase/database", () => {
    const data = { name: "unnamed" };
    
    const snapshot = { val: () => data };
    return {
      getDatabase: jest.fn(),
      onRegister: jest.fn(),
      initializeApp: jest.fn().mockReturnValue({
        database: jest.fn().mockReturnValue({
          ref: jest.fn().mockReturnThis(),
          once: jest.fn(() => Promise.resolve(snapshot))
        })
      })
    };
  });

  jest.mock("firebase/compat/app",()=>{
    return {
      initializeApp: jest.fn(),
      auth: jest.fn().mockReturnValue({
        onAuthStateChanged: jest.fn(),
        currentUser: true,
        signOut: jest.fn(),
        signInWithEmailAndPassword: jest.fn(),
        createUserWithEmailAndPassword: jest.fn()
      })
    };
  })


const value = {onLogout : null, user : {email: 'test@gmail.com', name: 'test'}};

test('test SettingsScreen component', () => {
    const element = create(
        
        <AuthenticationContext.Provider value={value}>
            <SettingsScreen />
        </AuthenticationContext.Provider>
      );

    expect(element.toJSON()).toMatchSnapshot();

  });