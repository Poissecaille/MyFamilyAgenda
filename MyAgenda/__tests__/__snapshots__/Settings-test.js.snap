// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SettingsScreen component 1`] = `
<RCTSafeAreaView
  color="#4A6EBF"
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#4A6EBF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#6200ee",
            "borderRadius": 90,
            "height": 180,
            "width": 180,
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {},
        ]
      }
    >
      <Text
        accessibilityElementsHidden={true}
        allowFontScaling={false}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        selectable={false}
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 108,
            },
            Array [
              Object {
                "lineHeight": 108,
                "transform": Array [
                  Object {
                    "scaleX": 1,
                  },
                ],
              },
              Object {
                "backgroundColor": "transparent",
              },
            ],
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰋦
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginTop": 16,
          },
        ]
      }
      variant="marginTop:16px"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#C8D6F6",
              "flexWrap": "wrap",
              "fontFamily": "Lato_400Regular",
              "fontSize": 16,
              "fontWeight": "500",
              "marginBottom": 0,
              "marginTop": 0,
            },
          ]
        }
        variant="label"
      >
        test@gmail.com
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "marginVertical": 8,
        },
        undefined,
      ]
    }
    theme={
      Object {
        "animation": Object {
          "scale": 1,
        },
        "colors": Object {
          "accent": "#03dac4",
          "backdrop": "rgba(0, 0, 0, 0.5)",
          "background": "#f6f6f6",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "error": "#B00020",
          "notification": "#f50057",
          "onSurface": "#000000",
          "placeholder": "rgba(0, 0, 0, 0.54)",
          "primary": "#6200ee",
          "surface": "#ffffff",
          "text": "#000000",
        },
        "dark": false,
        "fonts": Object {
          "light": Object {
            "fontFamily": "System",
            "fontWeight": "300",
          },
          "medium": Object {
            "fontFamily": "System",
            "fontWeight": "500",
          },
          "regular": Object {
            "fontFamily": "System",
            "fontWeight": "400",
          },
          "thin": Object {
            "fontFamily": "System",
            "fontWeight": "100",
          },
        },
        "roundness": 4,
      }
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          Array [
            Object {
              "padding": 8,
            },
            Array [
              Object {
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              },
              Object {
                "padding": 16,
              },
            ],
          ],
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "margin": 8,
                "width": 40,
              },
              Object {
                "marginLeft": 0,
                "marginRight": 16,
                "marginVertical": 0,
              },
            ]
          }
        >
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "lineHeight": 24,
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ],
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            󰠚
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "marginVertical": 6,
                "paddingLeft": 8,
              },
              Object {
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "fontSize": 16,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                  },
                  undefined,
                ],
              ]
            }
          >
            Logout
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
